### PR TITLE
TC improvements

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

- Show DEXs separate from CEXs in Exchange Settings
- Allow users to prefer DEXs for swaps
- Make swap Powered By tappable to go to Exchange Settings
- Change "Stake" to "Earn" in TransactionList
- Show APY value in TransactionList
- Show staked amount in TransactionList

Screen Shot 2023-01-03 at 02 25 14
Screen Shot 2023-01-03 at 02 24 37
Screen Shot 2023-01-03 at 02 24 29
Screen Shot 2023-01-03 at 02 24 16

### Dependencies

https://github.com/EdgeApp/edge-core-js/pull/495
https://github.com/EdgeApp/edge-exchange-plugins/pull/252

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203632680642545

### CODE REVIEW

File: src/components/scenes/CryptoExchangeQuoteScene.tsx [hunk 1]
```diff
@@ -3,12 +3,14 @@ import { EdgeAccount } from 'edge-core-js/types'
 import * as React from 'react'
 import { ScrollView, View } from 'react-native'
 import FastImage from 'react-native-fast-image'
+import { TouchableOpacity } from 'react-native-gesture-handler'
```
Comment (swansontec) [inline src/components/scenes/CryptoExchangeQuoteScene.tsx:6]: This is the wrong `TouchableOpacity`. I think you want the one from React Native itself. Easy mistake to make with auto-import.

**Review by swansontec — CHANGES_REQUESTED** [review]
This PR looks pretty solid in general, but I did find a few smaller things that might be worth looking at.

File: src/components/scenes/CryptoExchangeQuoteScene.tsx (hunk 1 above)
Comment (paullinator) [reply src/components/scenes/CryptoExchangeQuoteScene.tsx:6]: ack. good catch. I've become too reliant on Skynet

File: src/components/themed/TransactionListTop.tsx [hunk 2]
```diff
@@ -111,13 +113,45 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
...
+    const amount = positions.filter(p => p.currencyCode === currencyCode && p.pluginId === pluginId).reduce((prev, curr) => add(prev, curr.nativeAmount), '0')
+    return amount
+  }
+
+  updatePositions = async ({ stakePlugins = [], stakePolicies = [] }: { stakePlugins?: StakePlugin[]; stakePolicies?: StakePolicy[] }) => {
+    let lockedNativeAmount = '0'
+    for (const stakePolicy of stakePolicies) {
+      const { stakePolicyId } = stakePolicy
+      const stakePlugin = getPluginFromPolicy(stakePlugins, stakePolicy)
+      if (stakePlugin != null) {
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/themed/TransactionListTop.tsx:138]: 👍

**Review by swansontec — APPROVED** [review]

**Pull request merged**